cv_bridge/src/cv_bridge.cpp

```diff
@@ -409,14 +409,49 @@ cv::Mat matFromImage(const sensor_msgs::CompressedImage& source)
     return bgrMat;
 }
 
-sensor_msgs::CompressedImagePtr CvImage::toCompressedImageMsg(const std::string& dst_format) const
+sensor_msgs::CompressedImagePtr CvImage::toCompressedImageMsg(const Format dst_format) const
 {
   sensor_msgs::CompressedImagePtr ptr = boost::make_shared<sensor_msgs::CompressedImage>();
   toCompressedImageMsg(*ptr,dst_format);
   return ptr;
 }
 
-void CvImage::toCompressedImageMsg(sensor_msgs::CompressedImage& ros_image, const std::string& dst_format) const
+std::string getFormat(Format format) {
+
+	switch (format) {
+		case DIP:
+			return "dip";
+		case BMP:
+			return "bmp";
+		case JPG:
+			return "jpg";
+		case JPEG:
+			return "jpeg";
+		case JPE:
+			return "jpe";
+		case JP2:
+			return "jp2";
+		case PNG:
+			return "png";
+		case PBM:
+			return "pbm";
+		case PGM:
+			return "pgm";
+		case PPM:
+			return "ppm";
+		case RAS:
+			return "ras";
+		case SR:
+			return "sr";
+		case TIF:
+			return "tif";
+		case TIFF:
+			return "tiff";
+	}
+	return "";
```

Comment:
It's more that the behavior is undefined right ? What if I put an int that is bigger than the biggest enum in Format ? You can also have a warning message that says the format is not supported and default to JPG.